pages/scenarios/[runId].vue

```diff
@@ -120,56 +121,82 @@ import type { Parameter } from "~/types/parameterTypes";
 // TODO: Use the runId from the route rather than getting it out of the store.
 const appStore = useAppStore();
 
-const jobTakingLongTime = ref(false);
-const stoppedPolling = ref(false);
+let statusInterval: NodeJS.Timeout;
+const jobSlow = ref(false);
+const jobReallySlow = ref(false);
+const secondsSinceFirstStatusPoll = ref("0");
+
 const showSpinner = computed(() => {
   return (!appStore.currentScenario.result.data
-    && appStore.currentScenario.result.fetchStatus !== "error"
-    && !stoppedPolling.value);
+    && appStore.currentScenario.result.fetchStatus !== "error");
 });
 
 const paramDisplayText = (param: Parameter) => {
   if (appStore.currentScenario?.parameters && appStore.currentScenario?.parameters[param.id]) {
     const rawVal = appStore.currentScenario.parameters[param.id].toString();
+
+    const rawValIsNumberString = Number.parseInt(rawVal).toString() === rawVal;
+    if (rawValIsNumberString) {
+      // TODO: Localize number formatting.
+      return new Intl.NumberFormat().format(Number.parseInt(rawVal));
+    }
     return param.options ? param.options.find(({ id }) => id === rawVal)!.label : rawVal;
   }
 };
 
+// Use useAsyncData to store the time once, during server-side rendering: avoids client render re-writing value.
+const { data: timeOfFirstStatusPoll } = await useAsyncData<number>("timeOfFirstStatusPoll", async () => {
+  return new Promise<number>((resolve) => {
+    resolve(new Date().getTime());
+  });
+});
+
```

Comment:
Maybe just try declaring and setting directly in the setup tag - I *think* it should only get calculated once, on the server, but may be wrong!
`const timeOfFirstStatusPoll = new Date().getTime();`

Reply:
So what happens is this:

![image](https://github.com/user-attachments/assets/4cbafe89-bcd5-4c84-b6f6-932782fcbd01)

Here you see logs from one initial load of the 'new' page, as when you refresh that page or navigate to it from another site. This is rendered server side first, and then the setup script is run again on the client-side.

Most of the time (and indeed ALL of the time until we add the ability to refresh the results page), people will navigate to the results page from another page rather than loading it directly or refreshing it. And in _those_ cases, the setup script is run only on the client side:

![image](https://github.com/user-attachments/assets/3061d136-c296-4d24-aaaa-6f77421e56a7)

Finally, here are some logs of when we use useAsyncData when doing an initial load of the 'new' page. You can see that Nuxt retains the original value.
![image](https://github.com/user-attachments/assets/07890925-c996-4b9e-a7e9-f372d09969b2)

You can also inspect that value from the 'payload' section of the Nuxt DevTools:

![image](https://github.com/user-attachments/assets/a7f0ab36-daac-418c-a774-2ce7ce078372)

or by doing "View page source" from the browser (i.e. the originally loaded HTML, not the current HTML):

![image](https://github.com/user-attachments/assets/46d3d6ba-6714-4ae0-bb4f-f3aa4a233e61)